src/fsharp/NameResolution.fs

```diff
@@ -2483,11 +2483,78 @@ let ResolveFieldPrim (ncenv:NameResolver) nenv ad typ (mp,id:Ident) =
         if isAppTy g typ then 
             match ncenv.InfoReader.TryFindRecdOrClassFieldInfoOfType(id.idText,m,typ) with
             | Some (RecdFieldInfo(_,rfref)) -> [ResolutionInfo.Empty, FieldResolution(rfref,false)]
-            | None -> error(Error(FSComp.SR.nrTypeDoesNotContainSuchField((NicePrint.minimalStringOfType nenv.eDisplayEnv typ), id.idText),m))
+            | None ->
+                // record label doesn't belong to record type
+                // predict other labels of same record
+                let typeName = NicePrint.minimalStringOfType nenv.eDisplayEnv typ
+                
+                let labelsOfPossibleRecord =
+                    nenv.eFieldLabels
+                    |> Seq.filter (fun kv -> 
+                        kv.Value 
+                        |> List.map (fun r -> r.TyconRef.ToString()) // TODO: Ask Don how to compare type names here
```

Comment:
/cc @dsyme I think we want to compare full qualified names of the types. How can I do that?!